portal-impl/src/com/liferay/portal/service/impl/UserLocalServiceImpl.java

```diff
@@ -4827,9 +4824,9 @@ public User updateIncompleteUser(
 			}
 
 			validate(
-				companyId, user.getUserId(), autoPassword, password1, password2,
-				autoScreenName, screenName, emailAddress, null, firstName,
-				middleName, lastName, null, locale);
+				companyId, 0, user.getUserId(), autoPassword, password1,
+				password2, autoScreenName, screenName, emailAddress, null,
+				firstName, middleName, lastName, null, locale);
```

Comment:
Why do we have two magic values for the ldapServerId? If possible we should stick with one and use it consistently.

Reply:
Some users like the default user and anonymous users have an LDAP server Id of 0.  But, I see these validate methods either explicitly pass in a 0 (now -1) or they are for a user being added, so always using -1 is safe and more understandable.